cmd/registry/controller/controller.go

```diff
@@ -31,6 +31,20 @@ type Action struct {
 	RequiresReceipt   bool
 }
 
+func ValidateManifest(ctx context.Context, parent string, manifest *rpc.Manifest) bool {
+	isValid := true
+	for _, resource := range manifest.GeneratedResources {
+		log.FromContext(ctx).Debugf("Validating entry: %v", resource)
+		if errs := validateGeneratedResourceEntry(parent, resource); errs != nil {
```

Comment:
We should always consider `nil` slices the same as an empty slice to avoid confusion and subtle bugs, so we shouldn't be checking `errs != nil` here. Length checks are preferred.